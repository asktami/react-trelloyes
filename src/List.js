import React from 'react';
import { Card } from './Card';
import './List.css';

{
	/*
<section class="List">
				<header class="List-header">
					<h2>First list</h2>
				</header>
				<div class="List-cards">
					<div class="Card">
						<button type="button">delete</button>
						<h3>First card</h3>
						<p>lorem ipsum</p>
					</div>
					<button type="button" class="List-add-button">
						+ Add Random Card
					</button>
				</div>
</section>
*/
}

export class List extends React.Component {
	render() {
		const allCards = this.props.cards.map(card => (
			<Card key={card.id} title={card.title} content={card.content} />
		));

		return (
			<section className="List">
				<header className="List-header">
					<h2>{this.props.header}</h2>
				</header>
				<div className="List-cards">
					{allCards}
					<button type="button" className="List-add-button">
						+ Add Random Card
					</button>
				</div>
			</section>
		);
	}
}
