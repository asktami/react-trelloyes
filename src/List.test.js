// make React available
import React from 'react';

//make the ReactDOM available, necessary for rendering the component
import ReactDOM from 'react-dom';

// add snapshot tests
// first cd PROJECT to folder and do:
// npm install react-test-renderer -D
import renderer from 'react-test-renderer';

//make the List component available
import { List } from './List';

//this is the test suite
describe('List component tests', () => {
	// can also test with an empty array:
	// const myCards = [];

	const myCards = [
		{ id: 'a', title: 'First card', content: 'lorem ipsum' },
		{ id: 'b', title: 'Second card', content: 'lorem ipsum' },
		{ id: 'c', title: 'Third card', content: 'lorem ipsum' }
	];

	// smoke test:
	it('renders without crashing', () => {
		// first create a DOM element to render the component into
		const div = document.createElement('div');

		//render the component, this is the actual test, if something is wrong it will fail here
		ReactDOM.render(<List cards={myCards} />, div);

		//clean up code
		ReactDOM.unmountComponentAtNode(div);
	});

	// snapshot testing:
	/*
	1.  Render the component and create a human readable JSON file
   	2. Compare the rendered component to a saved version of the component
	*/

	// add a test to perform a snapshot test on the List component with the header prop set to myHeader and the cards set to a new array
	it('renders the UI as expected with cards array', () => {
		const tree = renderer
			.create(
				<section className="List">
					<header className="List-header">
						<h2>header={'myHeader'}</h2>
					</header>
					<div className="List-cards">
						<List cards={myCards} />
						<button type="button" className="List-add-button">
							+ Add Random Card
						</button>
					</div>
				</section>
			)
			.toJSON();
		expect(tree).toMatchSnapshot();
	});
});
