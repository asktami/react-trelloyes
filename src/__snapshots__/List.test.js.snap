// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`List component tests renders the UI as expected with cards array 1`] = `
<section
  className="List"
>
  <header
    className="List-header"
  >
    <h2>
      header=
      myHeader
    </h2>
  </header>
  <div
    className="List-cards"
  >
    cards=
    <button
      className="List-add-button"
      type="button"
    >
      + Add Random Card
    </button>
  </div>
</section>
`;
